### ISSUE TITLE
segmentation fault at start on Fedora 29

### ISSUE THREAD

**Issue by ziggy42** [shiftkey/desktop#78]
## Description
Trying to run desktop on Fedora 29 fails:
```bash
➜  ~ "/opt/GitHubDesktop/desktop" %U
[1]    21970 segmentation fault (core dumped)  "/opt/GitHubDesktop/desktop" %U
```

## Version
* GitHub Desktop: 1.4.2 Linux RC1 installed via `rpm`
* Operating system: Fedora 29

## Additional Information
I can see the same error for [Slack](https://stackoverflow.com/questions/53084955/why-slack-returns-a-segmentation-fault-after-fedora-29-upgrade): "It seems the issue is inside an "outdated" version of libnode.so."

**Comment by shiftkey** [comment]
My gut feeling is that this is related to older versions of Electron meshing with newer version of glibc, like https://github.com/electron/electron/issues/13972 - `1.4.2` does use Electron `2.0.9` which seems to have had something changed around it, but maybe it's still unresolved.

**Comment by ziggy42** [comment]
I bet it's fun packaging stuff for Linux, right? :smile:

**Comment by shiftkey** [comment]
![](https://media.giphy.com/media/xUOxf4YpQNlEtssg8g/giphy.gif)

**Comment by shiftkey** [comment]
@ziggy42 I've published [`release-1.5.0-linux2`](https://github.com/shiftkey/desktop/releases/tag/release-1.5.0-linux2) that includes the Electron 2.0.13 upgrade - let me know if that changes the behaviour on Fedora 29.

**Comment by ziggy42** [comment]
@shiftkey yes, it does, thank you!

PS:
this update changes the `.desktop` file apparently so right after install I got 2 icons for Github desktop. Fixed by:
```shell
sudo rm /usr/share/applications/desktop.desktop
```

PS 2:
I'm on Fedora so for the time being I'll choose `flatpak` > `rpm` > `snap`, but for your own sanity I'd say you could focus only on `snap`. Canonical is pushing it very well with apps like Spotify, slack, vs code etc all using it, and it'll be easier for you to have only 1 linux thing to support.
Plus I'm sure the guys at Canonical would very match love to help you with these.
